File: sdk/arena-java-sdk/src/main/java/com/github/kubeflow/arena/model/training/SparkJobBuilder.java
package com.github.kubeflow.arena.model.training;

import com.github.kubeflow.arena.enums.TrainingJobType;
import com.github.kubeflow.arena.exceptions.ArenaException;
import com.github.kubeflow.arena.model.fields.Field;
import com.github.kubeflow.arena.model.fields.StringField;

import java.util.ArrayList;
import java.util.Map;

public class SparkJobBuilder {

    protected String jobName;

    protected TrainingJobType jobType;

    protected ArrayList<Field> options;

    protected String command;

    public SparkJobBuilder() {
        this.jobType = TrainingJobType.SparkTrainingJob;
        this.options =  new ArrayList<Field>();
    }

    public SparkJobBuilder name(String name) {
        this.options.add(new StringField("--name",name));
        return this;
    }

    public SparkJobBuilder image(String image) {
        this.options.add(new StringField("--image",image));
        return this;
    }

    public SparkJobBuilder replicas(int replicas) {
        this.options.add(new StringField("--replicas",String.valueOf(replicas)));
        return this;
    }

    public SparkJobBuilder mainClass(String mainClass) {
        this.options.add(new StringField("--main-class",mainClass));
        return this;
    }

    public SparkJobBuilder jar(String jar) {
        this.options.add(new StringField("--jar",jar));
        return this;
    }

    public SparkJobBuilder driverCPU(int count) {
        this.options.add(new StringField("--driver-cpu-request",String.valueOf(count)));
        return this;
    }

    public SparkJobBuilder driverMemory(int count) {
        this.options.add(new StringField("--driver-memory-request",String.valueOf(count)));
        return this;
    }


    public SparkJobBuilder executorMemory(int count) {
        this.options.add(new StringField("--executor-memory-request",String.valueOf(count)));
        return this;
    }

    public SparkJobBuilder executorCPU(int count) {
        this.options.add(new StringField("--executor-cpu-request",String.valueOf(count)));
        return this;
    }

    public SparkJobBuilder command(String command) {
        this.command = command;
        return this;
    }


    public TrainingJob build() throws ArenaException {
        ArrayList<String> args = new ArrayList<String>();
        for (int i = 0;i < this.options.size();i++) {
            Field f = this.options.get(i);
            f.validate();
            for(int j = 0;j < f.options().size();j++) {
                args.add(f.options().get(j));
            }
        }
        return new TrainingJob(this.jobName,this.jobType,args,this.command);
    }
}

File: sdk/arena-python-sdk/arenasdk/training/spark_job_builder.py
#!/usr/bin/env python
from __future__ import annotations
from arenasdk.enums.types import TrainingJobType
from arenasdk.fields.fields import *
from arenasdk.training.job_builder import JobBuilder
from arenasdk.exceptions.arena_exception import ArenaException
from arenasdk.enums.types import ArenaErrorType
from arenasdk.training.job import TrainingJob

class SparkJobBuilder(object):
    def __init__(self):
        self._job_type: TrainingJobType = TrainingJobType.SparkTrainingJob
        self._job_name = ""
        self._options: List[Field] = list()
        self._command = ""
        
    def build(self) -> TrainingJob:
        args = list()
        try:
            for field in self._options:
                if not isinstance(field,Field):
                    raise ArenaException(ArenaErrorType.Unknown,"the type of option {} is not Field".format(field))
                field.validate()
                for opt in field.options():
                    args.append(opt)
            return TrainingJob(self._job_name,self._job_type,args,self._command)
        except ArenaException as e:
            raise e         
        
    def with_name(self,name: str) -> SparkJobBuilder:
        self._options.append(StringField("--name",name))
        return self 

    def with_image(self,image: str) -> SparkJobBuilder:
        self._options.append(StringField("--image",image))
        return self 

    def with_replicas(self,replicas: int) -> SparkJobBuilder:
        self._options.append(StringField("--replicas",str(replicas)))
        return self 

    def with_main_class(self,main_class: str) -> SparkJobBuilder:
        self._options.append(StringField("--main-class",main_class))
        return self 

    def with_jar(self,jar: str) -> SparkJobBuilder:
        self._options.append(StringField("--jar",jar))
        return self 

    def with_driver_cpu(self,driver_cpu: str) -> SparkJobBuilder:
        self._options.append(StringField("--driver-cpu-request",driver_cpu))
        return self 

    def with_driver_memory(self,driver_memory: str) -> SparkJobBuilder:
        self._options.append(StringField("--driver-memory-request",driver_memory))
        return self 
    
    def with_executor_memory(self,executor_memory: str) -> SparkJobBuilder:
        self._options.append(StringField("--executor-memory-request",executor_memory))
        return self 

    def with_executor_cpu(self,executor_cpu: str) -> SparkJobBuilder:
        self._options.append(StringField("--executor-cpu-request",executor_cpu))
        return self 

    def with_command(self,command: str) -> SparkJobBuilder:
        self._command = command
        return self
